GalaxyZooTouchTable/GalaxyZooTouchTable/Behaviors/UIElementDragBehavior.cs

```diff
@@ -0,0 +1,124 @@
+
+        private void AssociatedObject_TouchDown(object sender, TouchEventArgs e)
+        {
+            isTouchDown = true;
+        }
+
+        private void AssociatedObject_TouchLeave(object sender, TouchEventArgs e)
+        {
+            if (DragOverlay == null)
+            {
+                var visual = e.OriginalSource as Visual;
+                DragOverlay = (DragCanvas)UIElementDragBehavior.FindAncestor(typeof(DragCanvas), visual);
+            }
+
+            if (isTouchDown)
+            {
+                TouchPoint touchPosition = e.GetTouchPoint(DragOverlay);
+                Point initialPoint = new Point(touchPosition.Position.X, touchPosition.Position.Y);
+                FrameworkElement adornedElement = sender as FrameworkElement;
+
+                AdornerLayer adornerLayer = AdornerLayer.GetAdornerLayer(adornedElement);
+                GalaxyAdorner adorner = new GalaxyAdorner(adornedElement, initialPoint);
+                adornerLayer.Add(adorner);
+
+                EventHandler<TouchEventArgs> moveHandler = new EventHandler<TouchEventArgs>((s, evt) => DragDropContainer_TouchMove(s, e, adorner));
+                EventHandler<TouchEventArgs> upHandler = new EventHandler<TouchEventArgs>((s, evt) => DragDropContainer_TouchUp(s, evt, adorner, adornedElement));
+                EventHandler<TouchEventArgs> unsubscribeHandler = new EventHandler<TouchEventArgs>((s, evt) => DragDropContainer_Unsubscribe(s, e, moveHandler, upHandler, adorner));
+                adorner.UnsubscribeEvent = unsubscribeHandler;
+
+                DragOverlay.TouchMove += moveHandler;
+                DragOverlay.TouchUp += upHandler;
+                DragOverlay.TouchUp += unsubscribeHandler;
+            }
+            isTouchDown = false;
+        }
+
+        private void DragDropContainer_Unsubscribe(object s, TouchEventArgs e, EventHandler<TouchEventArgs> moveHandler, EventHandler<TouchEventArgs> upHandler, GalaxyAdorner adorner)
+        {
+            if (adorner.Parent == null)
+            {
+                DragOverlay.TouchMove -= moveHandler;
+                DragOverlay.TouchUp -= upHandler;
+                DragOverlay.TouchUp -= adorner.UnsubscribeEvent;
+            }
+        }
+
+        private void DragDropContainer_TouchMove(object sender, TouchEventArgs e, GalaxyAdorner adorner)
+        {
+            var touchPosition = e.GetTouchPoint(DragOverlay);
+            var adornerLocation = new Point(touchPosition.Position.X, touchPosition.Position.Y);
+            adorner.UpdatePosition(adornerLocation);
+        }
+
+        private void DragDropContainer_TouchUp(object sender, TouchEventArgs e, GalaxyAdorner adorner, UIElement element)
+        {
+            var touchPosition = e.GetTouchPoint(DragOverlay);
+            Point touchPoint = new Point(touchPosition.Position.X, touchPosition.Position.Y);
+
+            if (touchPoint.X >= adorner.Location.X && touchPoint.X <= adorner.Location.X + adorner.ActualWidth
```

Comment:
It may make sense to assign this boolean check to a variable just to better describe what you are checking